fe/fe-core/src/main/java/com/starrocks/load/DeleteHandler.java

```diff
@@ -901,6 +901,10 @@ public void replayDelete(DeleteInfo deleteInfo, GlobalStateMgr globalStateMgr) {
         if (deleteInfo == null) {
             return;
         }
+        if (isDeleteInfoExpired(deleteInfo, System.currentTimeMillis())) {
```

Comment:
You should add this check after updateTableDeleteInfo(), because it will update some flag of its related table.

Reply:
Done.